### ISSUE TITLE
Coin displays "In Mempool" even after 6 confirmations

### ISSUE THREAD

**Issue by cbjd** [layer2tech/mercury-wallet#876]
Coin was deposited and has received six confirmations, but still shows "In Mempool." Block height is current and connected to all servers. Attempted to close and re-open app as well. 

Edit: I restored the seed as a new wallet and it was able to be spent, but then I exited and reopened it, and now it shows in the new wallet as "expired," in the other wallet it still shows as "In Mempool."

Edit 2: When I load the original wallet, I now see this error in the console: 

```{name: "StatusCodeError", statusCode: 400,…}
error: "sendrawtransaction RPC error: {"code":-27,"message":"Transaction already in block chain"}"
message: "400 - "sendrawtransaction RPC error: {\"code\":-27,\"message\":\"Transaction already in block chain\"}""
name: "StatusCodeError
```

Statecoin address: sc1qgvqes569uu3gp9j2n38hzzy6rway804avjf2qmtx4g6kv2wv9zzv2q6hz9

Mercury .4.5.0
OS X 10.15.7

**Comment by tomt1664** [comment]
Looking into this now, and trying to find out exactly what went wrong and how to prevent it in future. But what appears to have happened is that (probably due to a connection problem connecting to the electrum server and getting the current blockheight correctly) that the wallet set the locktime (block height) of the backup transaction (that enables you to recover the funds even if the mercury server disappears) incorrectly. 

The wallet _should_ take the current blockheight and add the expiry time (in blocks) of 12960 (3 months) and then create the backup tx to be co-signed. What did happen is that the current blockheight was set to 0 (in error) and so the locktime was set to 0+12960, which is in the past. 

This means that the coin became 'expired' immediately, and the backup tx was broadcast and has confirmed (as would happen if any coin reaches the expiry blockheight). The backup tx is here: https://blockstream.info/tx/b6b64d387184f3bdf6ffe5355779c8fc8c08b89914e8090ddb26d04d24d619e8

Your recovered wallet should have full control of the coin. You can access the private key to the coin by going to settings (gear icon in the tip right corner) and then clicking on 'Manage backup transactions'. Select the coin, and you can view the private key. Click 'Show' to copy the private key (in WIF format) which can be imported into most other bitcoin wallets. 

![Screenshot 2021-11-10 at 18 48 33](https://user-images.githubusercontent.com/15906918/141683433-6684b002-41c4-49d0-bfa8-041a2bcd2856.png)

Let us know if this doesn't work. If you want, send us an address or LN invoice and we'll refund the fee.

**Comment by cbjd** [comment]
I was able to export the private key and sweep the funds from it.

Should the status of the coin change in the wallet from Expired to Withdrawn once I have swept it?

**Comment by tomt1664** [comment]
Seems it doesn't at the moment. This should be straightforward to do - we'll add this to the next release (along with logic to prevent the incorrect locktime being set). Thanks.

**Comment by tomt1664** [comment]
Fixed in 0.4.50
